api/projects/serializers.py

```diff
@@ -55,6 +55,15 @@ def get_use_edge_identities(self, obj: Project) -> bool:
             == ProjectIdentityMigrationStatus.MIGRATION_COMPLETED.value
         )
 
+    def validate(self, data):
+        subscription_metadata = data[
+            "organisation"
+        ].subscription.get_subscription_metadata()
+        total_projects = len(Project.objects.filter(organisation=data["organisation"]))
+        if subscription_metadata.projects <= total_projects:
+            raise serializers.ValidationError("Max projects reached for subscription")
+        return data
```

Comment:
I'd very much prefer to see this logic in the `ProjectPermissions` class (api/projects/permissions.py).

Reply:
Sure, I have moved it to ProjectPermissions and added some tests as well.